toolbox/src/base/WBBlock.cpp

```diff
@@ -0,0 +1,275 @@
+        return false;
+    }
+
+    std::vector<std::string> controlBoardsNames;
+    for (auto cell: cellCBN) {
+        std::string joint;
+        if (!cell->asString(joint)) {
+            Log::getSingleton().Error("Failed to convert ControlBoardsNames from cell to string.");
+            return false;
+        }
+        controlBoardsNames.push_back(joint);
+    }
+
+    // LocalName
+    std::string localName;
+    if (!s["LocalName"]->asString(localName)) {
+        Log::getSingleton().Error("Cannot retrieve string from LocalName parameter.");
+        return false;
+    }
+
+    std::vector<double> gravityVector;
+    if (!s["GravityVector"]->asVectorDouble(gravityVector)) {
+        Log::getSingleton().Error("Cannot retrieve vector from GravityVector parameter.");
+        return false;
+    }
+
+    // Create the ToolboxConfig object
+    // ===============================
+
+    // Populate a Property object that stores the input parameters
+    config.setRobotName(robotName);
+    config.setUrdfFile(urdfFile);
+    config.setControlledJoints(controlledJoints);
+    config.setControlBoardsNames(controlBoardsNames);
+    config.setLocalName(localName);
+    config.setGravityVector(gravityVector);
+
+    return true;
+}
+
+const Configuration& WBBlock::getConfiguration()
+{
+    ToolboxSingleton& interface = ToolboxSingleton::sharedInstance();
+    return interface.getConfiguration(confKey);
+}
+
+const std::shared_ptr<RobotInterface> WBBlock::getRobotInterface()
+{
+    ToolboxSingleton& interface = ToolboxSingleton::sharedInstance();
+    return interface.getRobotInterface(confKey);
+}
+
+bool WBBlock::configureSizeAndPorts(BlockInformation* blockInfo)
+{
+    // Remember not allocate / save any data in this function, it won't be persistent.
+
+    // Setup the ToolboxSingleton
+    // ==========================
+
+    // Setup the DOFs (required by childs for configuring port sizes)
+    // --------------------------------------------------------------
+
+    // Initialize the ToolboxSingleton
+    ToolboxSingleton& interface = ToolboxSingleton::sharedInstance();
+
+    // Get the absolute name of the block from which the configuration has been retrieved.
+    // This will be the key of the map containing all the configurations of the ToolboxSingleton.
+    if (!blockInfo->getStringParameterAtIndex(PARAM_IDX_CONFBLOCKNAME, confKey)) {
+        Log::getSingleton().Error("Failed to retrieve the struct with parameters.");
+        return false;
+    }
+
+    // Get the configuration struct from the block
+    Configuration config;
+    if (!getWBToolboxParameters(config, blockInfo)) {
+        return false;
+    }
+
+    // Check if the configuration is valid
+    if (!config.isValid()) {
+        Log::getSingleton().Error("The provided configuration is not valid.");
+        return false;
+    }
+
+    // Store the configuration inside the ToolboxSingleton
+    if (!interface.configure(confKey, config)) {
```

Comment:
I don't like `configure` as it seems it configure something, while actually it is just saving the configuration, am I right?

Reply:
Replied above